seatunnel-connectors-v2/connector-databend/src/main/java/org/apache/seatunnel/connectors/seatunnel/databend/sink/DatabendSinkWriter.java

```diff
@@ -110,46 +143,142 @@ public DatabendSinkWriter(
                         e);
             }
+        }
+    }
 
-            this.insertSql = generateInsertSql(database, table, rowType);
-            log.info("Generated insert SQL: {}", insertSql);
-            try {
-                this.schemaChangeManager = new SchemaChangeManager(databendSinkConfig);
-                this.preparedStatement = connection.prepareStatement(insertSql);
-                this.preparedStatement.setQueryTimeout(executeTimeoutSec);
-                log.info("PreparedStatement created successfully");
-            } catch (SQLException e) {
-                throw new DatabendConnectorException(
-                        DatabendConnectorErrorCode.SQL_OPERATION_FAILED,
-                        "Failed to prepare statement: " + e.getMessage(),
-                        e);
-            }
+    private String getCurrentTimestamp() {
+        return LocalDateTime.now().format(DateTimeFormatter.ofPattern("yyyyMMddHHmmss"));
+    }
+
+    private void initializeCdcPreparedStatement() throws SQLException {
+        log.info("Initializing CDC PreparedStatement");
+
+        // In CDC mode, the rawTableName should be set by the AggregatedCommitter
+        // If it's not set yet, we can't proceed with CDC operations
+        if (rawTableName == null || rawTableName.isEmpty()) {
+            throw new DatabendConnectorException(
+                    DatabendConnectorErrorCode.SQL_OPERATION_FAILED,
+                    "Raw table name not set by AggregatedCommitter. Cannot initialize CDC PreparedStatement.");
+        }
+
+        // Generate insert SQL for raw table
+        String insertRawSql = generateInsertRawSql(sinkTablePath.getDatabaseName());
+
+        // Create the PreparedStatement
+        this.cdcPreparedStatement = connection.prepareStatement(insertRawSql);
+        this.cdcPreparedStatement.setQueryTimeout(executeTimeoutSec);
+
+        log.info("CDC PreparedStatement created successfully with SQL: {}", insertRawSql);
+    }
+
+    private void initTraditionalMode(String database, String table) throws SQLException {
+        // use the catalog table schema to create the target table
+        SeaTunnelRowType rowType = catalogTable.getSeaTunnelRowType();
+        if (rowType == null || rowType.getFieldNames().length == 0) {
+            throw new DatabendConnectorException(
+                    DatabendConnectorErrorCode.SCHEMA_NOT_FOUND,
+                    "Source table schema is empty or null");
+        }
+
+        if (!tableExists(database, table)) {
+            log.info(
+                    "Target table {}.{} does not exist, creating with source schema",
+                    database,
+                    table);
+            createTable(database, table, rowType);
+        } else {
+            log.info("Target table {}.{} exists, verifying schema", database, table);
+            verifyTableSchema(database, table, rowType);
+        }
```

Comment:
we should create table in [`SaveModeHandler`](https://github.com/apache/seatunnel/blob/0369c7a8834ff1bd595b42203215e3fc11eadc8f/seatunnel-connectors-v2/connector-databend/src/main/java/org/apache/seatunnel/connectors/seatunnel/databend/sink/DatabendSink.java#L151). So I think this part is useless.

Reply:
Thanks, this part is for testing, I forgot to remove it. Removed.